pkg/cmd/server/kubernetes/node_config.go

```diff
@@ -148,6 +152,12 @@ func BuildKubernetesNodeConfig(options configapi.NodeConfig) (*NodeConfig, error
 	server.PodInfraContainerImage = imageTemplate.ExpandOrDie("pod")
 	server.CPUCFSQuota = true // enable cpu cfs quota enforcement by default
 
+	if dockerClient, _, err := docker.NewHelper().GetClient(); err == nil {
+		if docker.IsAPIVersionCompatible(dockerClient, minimumDockerAPIVersionForParallelPulls) {
```

Comment:
Is SerializeImagePulls *only* a workaround for Docker being broken in 1.8?
Is there ever any case an admin wants to force its use?


On Tue, Feb 2, 2016 at 10:00 AM, Avesh Agarwal <notifications@github.com>
wrote:

> In pkg/cmd/server/kubernetes/node_config.go
> <https://github.com/openshift/origin/pull/6538#discussion_r51578882>:
>
> > @@ -148,6 +152,12 @@ func BuildKubernetesNodeConfig(options configapi.NodeConfig) (*NodeConfig, error
> >  	server.PodInfraContainerImage = imageTemplate.ExpandOrDie("pod")
> >  	server.CPUCFSQuota = true // enable cpu cfs quota enforcement by default
> >
> > +	if dockerClient, _, err := docker.NewHelper().GetClient(); err == nil {
> > +		if docker.IsAPIVersionCompatible(dockerClient, minimumDockerAPIVersionForParallelPulls) {
>
> You mean a new config flag for setting docker version and then setting
> serializeimagepull based on that new flag? Somehow I think it could be more
> error prone even for admins, 1) what if the new flag and
> serializeimagepulls are configured incorrectly and not in accordance to
> each other, 2) what if docker is upgraded or downgraded and an admin
> forgets to reconfigure any of these flag.
>
> I was thinking more about it and I am still wondering why is it an issue
> to determine docker version at run time (infact to avoid manual errors) and
> then setting config vars accordingly? I am still trying to understand if
> there is any technical reason for that?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/6538/files#r51578882>.
>